### PR TITLE
[DOCS] Reformat `mapping` charfilter

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:

* Adds title abbreviation
* Adds Lucene link to description
* Add standard headings
* Simplifies analyze example
* Add contextual text and simplifies analyzer example

### CODE REVIEW

File: docs/reference/analysis/charfilters/mapping-charfilter.asciidoc [hunk 1]
```diff
@@ -8,75 +11,53 @@ with the value associated with that key.
 Matching is greedy; the longest pattern matching at a given point wins.
```
Comment (debadair) [inline docs/reference/analysis/charfilters/mapping-charfilter.asciidoc:11]: This is my favorite line of the day. :-)

**Review by debadair — APPROVED** [review]
LGTM

**Comment by jrodewig** [comment]
Thanks @debadair .

**Comment by jrodewig** [comment]
### Backport commits
master fd8af38 
7.x 24a50eb 
7.8 b37c2b6
7.7 3fa6dba

**Pull request merged**